### PR TITLE
Re-sizing the Situation - Exile's Reach

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**

-  Add quest for Alliance and Horde

**Tests performed:**

Builds and tested in game.

### CODE REVIEW

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 1]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+{
+    npc_lindie_springstock_q56034(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        Player* player = me->GetOwner()->ToPlayer();
+        if (!player)
+            return;
+
+        Conversation* conversation = Conversation::CreateConversation(12086, player, *player, player->GetGUID(), nullptr, false);
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5364]: magic number

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 2]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+Position ResizingGuardianPosition = { 100.56077f, -2418.0713f, 90.34765f };
+
+// Entry 156749 - Lindie Springstock
+struct npc_lindie_springstock_q56034 : public ScriptedAI
+{
+    npc_lindie_springstock_q56034(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        Player* player = me->GetOwner()->ToPlayer();
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5360]: crash if owner is null (e.g. via `.npc add temp`)

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 3]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+
+    void MovementInform(uint32 uiType, uint32 uiId) override
+    {
+        if (uiType != POINT_MOTION_TYPE)
+            return;
+
+        if (uiId != 0)
+            return;
+
+        if (Player* player = me->GetOwner()->ToPlayer())
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5384]: crash if owner is null (e.g. via `.npc add temp`), also casting to player is not required here

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 4]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+};
+
+// Entry 167915 - Cork Fizzlepop
+struct npc_cork_fizzlepop_q59941 : public ScriptedAI
+{
+    npc_cork_fizzlepop_q59941(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        Player* player = me->GetOwner()->ToPlayer();
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5432]: crash if owner is null (e.g. via `.npc add temp`)

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 5]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+        if (!player)
+            return;
+
+        Conversation* conversation = Conversation::CreateConversation(12086, player, *player, player->GetGUID(), nullptr, false);
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5436]: magic number

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 6]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+
+    void MovementInform(uint32 uiType, uint32 uiId) override
+    {
+        if (uiType != POINT_MOTION_TYPE)
+            return;
+
+        if (uiId != 0)
+            return;
+
+        if (Player* player = me->GetOwner()->ToPlayer())
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5457]: crash if owner is null (e.g. via `.npc add temp`), also casting to player is not required here

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 7]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+            player->RemoveAura(SPELL_SUMMON_CORK_FIZZLEPOP_GUARDIAN_Q59941);
+        }
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        if (spellInfo->Id != SPELL_PING_LINDIE_Q56034_Q59941)
+            return;
+
+        Player* player = me->GetOwner()->ToPlayer();
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5469]: crash if owner is null (e.g. via `.npc add temp`)

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 8]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+        if (uiType != POINT_MOTION_TYPE)
+            return;
+
+        if (uiId != 0)
+            return;
+
+        me->SetFacingTo(0.785398f);
+
+        if (Unit* owner = me->GetOwner())
+            if (owner->ToPlayer())
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5547]: is there a case where owner is != player? if not, remove this if

File: src/server/scripts/ExilesReach/zone_exiles_reach.cpp [hunk 9]
```diff
@@ -5330,6 +5330,562 @@ class quest_choppy_booster_mk5 : public quest_scout_chopper
...
+
+// Spell 305749 - Summon Admiral Garrick Guardian
+// Spell 326635 - Summon Cork (DNT)
+class spell_summon_guardian_q56034_q59941 : public SpellScript
+{
+    // @TODO: drop after TARGET_UNK_142 impl
+
+    void SelectTarget(WorldObject*& target)
+    {
+        Player* caster = GetCaster()->ToPlayer();
```
Comment (mdX7) [inline src/server/scripts/ExilesReach/zone_exiles_reach.cpp:5586]: is the spell also casted from non players?

**Review by mdX7 — COMMENTED** [review]
code seems fine, only found some minor issues! :)

**Comment by malcrom** [comment]
Needs new review

**Comment by mdX7** [comment]
fyi `spell_linked_spell` should no longer be used, write spell scripts instead. Given the spell is removed via script and the spell is already casted and i didnt see any difference ingame i just removed the `spell_linked_spell` lines

**Pull request merged**